GVFS/GVFS.Common/GVFSEnlistment.cs

```diff
@@ -220,6 +220,11 @@ public bool TryCreateEnlistmentFolders()
             try
             {
                 Directory.CreateDirectory(this.EnlistmentRoot);
+                if (!GVFSPlatform.Instance.FileSystem.IsFileSystemSupported(this.EnlistmentRoot, out string error))
+                {
+                   throw new InvalidRepoException($"FileSystem unsupported: {error}");

```

Comment:
It looks a bit strange to be throwing `InvalidRepoException` from inside a `Try` method that typically returns true/false.  

Could you either:

- Update the method to return an output error string and return false when IsFileSystemSupported returns false

-or-

- Move the call to `IsFileSystemSupported` up so that it's after the call to `TryCreateEnlistmentFolders` so that we can return a more specific error there?  In other words:

```
if (!enlistment.TryCreateEnlistmentFolders())
{
    string error = "Could not create enlistment directory";
    tracer.RelatedError(error);
    return new Result(error);
}

if (!enlistment.IsFileSystemSupported(this.EnlistmentRoot, out string error))
{
    string error = $"FileSystem unsupported: {error}";
    tracer.RelatedError(error);
    return new Result(error);
}

```